--- data/app/lib/private/Preview/PDF.php	2020-12-20 22:19:33.097808495 +0900
+++ ./PDF.php	2020-12-20 22:06:05.147818209 +0900
@@ -30,11 +30,35 @@
 use OC\Preview\GeneratorHelper;
 
 //.pdf
-class PDF extends Bitmap {
+class PDF extends ProviderV2 {
 	/**
 	 * {@inheritDoc}
 	 */
 	public function getMimeType(): string {
 		return '/application\/pdf/';
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getThumbnail(File $file, int $maxX, int $maxY): ?IImage {
+		// NOTE: uetchy
+		$tmpPath = $this->getLocalFile($file);
+		$outputPath = \OC::$server->getTempManager()->getTemporaryFile();
+
+		$gsBin = \OC_Helper::findBinaryPath('gs');
+		$cmd = $gsBin . " -o " . escapeshellarg($outputPath) . " -sDEVICE=pngalpha -dLastPage=1 " . escapeshellarg($tmpPath);
+		// shell_exec("echo " . $cmd . " 1>&2");
+		shell_exec($cmd);
+
+		$this->cleanTmpFiles();
+
+		//new bitmap image object
+		$image = new \OC_Image();
+		$image->loadFromFile($outputPath);
+		$image->scaleDownToFit($maxX, $maxY);
+		//check if image object is valid
+		unlink($outputPath);
+		return $image->valid() ? $image : null;
+	}
 }
